### ISSUE TITLE
utf16leSlice behaviour differs from node.js

### ISSUE THREAD

**Issue by roaringUnicorn** [feross/buffer#251]
Problem
```js
const oddLengthBuffer = Buffer.from('abcde', 'utf8')
console.log(Buffer.from(oddLengthBuffer.toString('utf16le'), 'utf16le').toString('utf8')) // ‘abcd’ in Node.js and ‘abcd\x00\x00’ with polyfill
```

This comes from accessing n+1-th element of a Buffer of size n in utf16leSlice.
Node.js simply omits it.

Proposed solution: https://github.com/feross/buffer/pull/250

**Comment by tex0l** [comment]
@feross I have the same issue, the suggested PR works, when do you plan on reviewing it?

**Comment by feross** [comment]
Released as 5.7.0.
